benchmarks/inference-server/triton/README.md

```diff
@@ -0,0 +1,198 @@
+
+   ```Dockerfile
+   FROM nvcr.io/nvidia/tritonserver:23.10-trtllm-python-py3
+
+   # Copy the necessary files into the container
+   COPY all_models /all_models
+   COPY scripts /opt/scripts
+
+   # Install Python dependencies
+   RUN pip install sentencepiece protobuf
+
+   RUN chmod +x /opt/scripts/start_triton_servers.sh
+
+   CMD ["/opt/scripts/start_triton_servers.sh"]
+   ```
+
+   The Shell script `/opt/scripts/start_triton_servers.sh` is like below: 
+
+   ```start_triton_servers.sh
+   #!/bin/bash
+
+
+    # Use the environment variable to log in to Hugging Face CLI
+    huggingface-cli login --token $HUGGINGFACE_TOKEN
+
+
+
+
+    # Launch the servers (modify this depending on the number of GPU used and the exact path to the model repo)
+    mpirun --allow-run-as-root -n 1 /opt/tritonserver/bin/tritonserver \
+    --model-repository=/all_models/inflight_batcher_llm \
+    --disable-auto-complete-config \
+    --backend-config=python,shm-region-prefix-name=prefix0_ : \
+    -n 1 /opt/tritonserver/bin/tritonserver \
+    --model-repository=/all_models/inflight_batcher_llm \
+    --disable-auto-complete-config \
+    --backend-config=python,shm-region-prefix-name=prefix1_ :```
+   ```
+   *Build and Push the Docker Image:*
+
+   Build the Docker image and push it to your container registry:
+
+   ```
+   docker build -t your_registry/tritonserver_llm:latest .
+   docker push your_registry/tritonserver_llm:latest
+
+   ```
+   Replace `your_registry` with your actual Docker registry path.
+
+   ***Method 2: Upload Model repository to gcs***
+
+   In this method we can directly upload the model engine to gcs and use the base image provided by Nvidia and specify the command to launch the triton server via the deployment yaml file: 
+   ```
+   gsutil cp -r your_model_folder gs://your_model_repo/all_models/ 
+   ```
+   Replace `your_model_repo` with your actual gcs repo path.
+
+
+
+
+## Step 2: Create and Configure `terraform.tfvars`
+
+1. **Initialize Terraform Variables:**
+
+   Create a `terraform.tfvars` file by copying the provided example:
+
+   ```bash
+   cp sample-terraform.tfvars terraform.tfvars
+   ```
+
+2. Define `template_path`, `image_path` and `gpu_count` variables  in `terraform.tfvars`.
+   
+   * If using method 1 In Step 1 above:
+   ```bash
+   template_path = "path_to_manifest_template/triton-tensorrtllm-inference.tftpl" 
+   image_path = "path_to_your_registry/tritonserver_llm:latest"
+   gpu_count = X
+   ```
+   * If using method 2 In Step 1 above:
+   
+   ```bash
+   template_path = "path_to_manifest_template/triton-tensorrtllm-inference-gs.tftpl"
+   image_path = "nvcr.io/nvidia/tritonserver:23.10-trtllm-python-py3"
+   gpu_count = X
+   ```
+   and also update the `triton-tensorrtllm-inference-gs.tftpl` with the path to the gcs repo under InitContainer and the command to launch the container under Container section.
```

Comment:
We should have a variable defined in the .tfvars file and pipe through the automation rather than asking customer to edit .tftpl files directly

Reply:
yes I thought about that but missed adding that. Makes sense.  let me define this in the .tfvars file.